lib/collection/src/wal_delta.rs

```diff
@@ -162,18 +198,987 @@ fn resolve_wal_delta(
+
+            a_wal.lock_and_write(&mut operation_a).await.unwrap();
+            b_wal.lock_and_write(&mut operation_b).await.unwrap();
+            b_wal
+                .lock_and_write(&mut operation_b_forward)
+                .await
+                .unwrap();
+
+            c_clock_0.advance_to(operation_a.clock_tag.unwrap().clock_tick);
+            c_clock_0.advance_to(operation_b.clock_tag.unwrap().clock_tick);
+        }
+
+        // Continue recovery
+        {
+            let op3 = mock_operation(3);
+            let op3_with_clock = OperationWithClockTag::new(op3, None);
+            b_wal
+                .lock_and_write(&mut op3_with_clock.clone())
+                .await
+                .unwrap();
+
+            let op30 = mock_operation(30);
+            let op30_with_clock = OperationWithClockTag::new(op30, None);
+            b_wal
+                .lock_and_write(&mut op30_with_clock.clone())
+                .await
+                .unwrap();
+
+            let op4 = mock_operation(4);
+            let op4_with_clock = OperationWithClockTag::new(op4, None);
+            b_wal
+                .lock_and_write(&mut op4_with_clock.clone())
+                .await
+                .unwrap();
+
+            let op5 = mock_operation(5);
+            let op5_with_clock = OperationWithClockTag::new(op5, None);
+            b_wal
+                .lock_and_write(&mut op5_with_clock.clone())
+                .await
+                .unwrap();
+
+            // Once full transfer is done, we update cutoff point on B to the last seen of A
+            b_wal.update_cutoff(&a_wal.recovery_point().await).await;
+        }
+
+        // Try to recover E from B
+        let e_recovery_point = e_wal.recovery_point().await;
+
+        // Cannot recover E from B, because B has a high cutoff point due to the full transfer
+        let delta_from = b_wal.resolve_wal_delta(e_recovery_point.clone()).await;
+        assert_eq!(delta_from.unwrap_err(), WalDeltaError::Cutoff);
+
+        // TODO: the following bit fails, because we cannot recover E from B here, since B has a
+        // TODO: high cutoff point due to the recent full transfer
+        {
+            let delta_from = b_wal
+                .resolve_wal_delta(e_recovery_point.clone())
+                .await
+                .unwrap()
+                .unwrap();
+
+            // Operation 0 was written to E, but everything else has to be written to E (including transfers)
+            assert_eq!(delta_from, 1);
+
+            // Total: 2 operations from C
+            // + 1 operation from C_1
+            // + 1 operations from D (one missing)
+            // + 5 recovery operations
+            // + 1 operation from C during recovery
+            // + 1 duplicate operation from C
+            assert_eq!(b_wal.wal.lock().read(delta_from).count(), 11);
+
+            // Recover E from B
+            e_wal.append_from(&b_wal, delta_from).await.unwrap();
+        }
```

Comment:
This bit is now problematic.

The intent here is to recover E from B using diff transfer, but we cannot do that. B has a high cutoff point because it just received a full transfer, so it cannot provide a delta big enough to recover E.

@generall Could you check how we can change this to match your original intent?

Reply:
Neither A or E can be recovered from B, I guess. That's fine, let's change the scenario